### PR TITLE
Update the arrivalTime with the `createdAt` time.

### PR DESCRIPTION
**Pull request by floitsch**
Overwrites the arrival time of the meta information with the "createdAt" time.

### CODE REVIEW

**Comment by floitsch** [comment]
Given that the Toit server can buffer messages and isn't guaranteed to maintain order, it feels like using the "createdAt" time for the "arrivalTime" makes some sense.

Alternatively, I could add this information as additional meta field.

File: packages/stream/toit/src/ToitConsumer.js [hunk 1]
```diff
@@ -50,11 +51,10 @@ module.exports = class ToitConsumer extends sdk.ConsumerMixin(ToitConnector) {
 					let msg = envelope.getMessage();
 					let createdAt = msg.getCreatedAt().toDate();
 					let data = Buffer.from(msg.getData(),  "utf-8");
-					let decoded = JSON.parse(data);
-					this.onMessage(this.config.topic, {
-						"data": decoded,
-						"createdAt": createdAt,
-					})
+					let meta = {
+						arrivalTime: ms2serial(createdAt),
```
Comment (cleancoderocker) [inline packages/stream/toit/src/ToitConsumer.js:55]: It would indeed be better to use another property instead of `arrivalTime`.

**Comment by TENHOUBEN** [comment]
Hi, I doubt that this would work, because the `arrivalTime` property is set within a machine too (when a message is added to the inbox). So I propose you better go with your suggested alternative, i.e. simply add an additional meta-field.

**Comment by floitsch** [comment]
> Hi, I doubt that this would work, because the `arrivalTime` property is set within a machine too (when a message is added to the inbox). So I propose you better go with your suggested alternative, i.e. simply add an additional meta-field.

I think I tested it, and was able to override the value. But not 100% sure.

Will go with an extra meta-field. (`createdAt`).

**Comment by TENHOUBEN** [comment]
Ok, I've checked a bit closer and you are right :-) 
Same fields of a received message overwrite those added by machine.
So both alternatives will do...

**Comment by TENHOUBEN** [comment]
Yes, serial-number is the preferred time format within streamsheets, because it makes time based calculation easier. However, since you use a custom property you are free to choose. But if you plan to use this property for calculation using a serial-number is probably better.

**Pull request merged**